core/lib/dal/src/server_notifier.rs

```diff
@@ -0,0 +1,76 @@
+use zksync_db_connection::{connection::Connection, error::DalResult, instrument::InstrumentExt};
+use zksync_types::{server_notification::ServerNotification, L1BlockNumber, H256};
+
+use crate::Core;
+
+#[derive(Debug)]
+pub struct ServerNotificationsDal<'a, 'c> {
+    pub storage: &'a mut Connection<'c, Core>,
+}
+
+impl ServerNotificationsDal<'_, '_> {
+    pub async fn save_notification(
+        &mut self,
+        main_topic: H256,
+        l1_block_number: L1BlockNumber,
+        value: serde_json::Value,
```

Comment:
- Nit: Why doesn't this use `ServerNotification`?
- What does `ServerNotification.value` supposed to mean? AFAIU, it's always set to `null` in the `eth_watch` code.

Reply:
It was an idea to migrate all server notification to this code. but looks like a premature optimization